File: python/resistor-color-trio/resistor_color_trio.py
BANDS = {   
        "black":   0,
        "brown":   1,
        "red":     2,
        "orange":  3,
        "yellow":  4,
        "green":   5,
        "blue":    6,
        "violet":  7,
        "grey":    8,
        "white":   9 
  }

def label(colors):
    value = str((10*BANDS[colors[0]] + BANDS[colors[1]]) * 10**BANDS[colors[2]])
    if value[-9:] == '000000000':
        return value[:-9] + ' gigaohms'
    elif value[-6:] == '000000':
        return value[:-6] + ' megaohms'
    elif value[-3:] == '000':
        return value[:-3] + ' kiloohms'
    else:
        return value + ' ohms'

File: typescript/resistor-color-trio/resistor-color-trio.ts
export const COLORS = ['black', 'brown', 'red', 'orange', 'yellow', 'green', 'blue', 'violet', 'grey', 'white']

export const colorCode = (color: string): number => {
  return COLORS.indexOf(color);
}

export const decodedValue = (colors: string[]): number => {
  const band1 = colorCode(colors[0]);
  const band2 = colorCode(colors[1]);
  const band3 = colorCode(colors[2]);
  const value = `${band1}${band2}e${band3}`;
  return Number(value);
}

export const decodedResistorValue = (colors: string[]): any => {
  let value = decodedValue(colors);
  
  if (value >= 1e9) {
    value /= 1e9
    return `${value} gigaohms`;
  } 
  if (value >= 1e6) {
    value /= 1e6
    return `${value} megaohms`;
  } 
  if (value >= 1e3) {
    value /= 1e3
    return `${value} kiloohms`;
  }
  return `${value} ohms`;
}
